### PR TITLE
Add local test env helper and mock pachd

### PR DESCRIPTION
**Pull request by Tryneus**
This PR adds:

* `testutil.Env` - a structure containing an etcd server/client pair, and a mock pachd with a pachd client
* `testutil.WithEnv` - a function that instantiates a `testutil.Env`, calls a callback, and cleans up the env afterwards.
* `testutil.MockPachd` - a GRPC server hosting a mock implementation of pachd, all calls error by default but can be overridden by test code
* `testutil.NewMockPachd` - a function to instantiate a new `MockPachd` listening on a randomly-assigned port

In addition, several changes were made to `grpcutil.Serve`, which I believe was broken.  It now:
* correctly handles multiple servers (it used to block on the first server created rather than run them in parallel)
* returns once all servers are ready (some users used the 'register' callback to determine readiness, which was rather hacky)
* returns an `errgroup.Group` to wait for the servers to finish
* uses a context for cancellation instead of an unused `Cancel` option
* allows specifying the hostname to listen on and the selection of a random port
* now performs orderly shutdown of the grpc servers rather than closing the listener out from under it (which would cause an error on the sever)
  * this is actually a breaking change, although it seems right to me - we could use `server.Stop()` instead of `server.GracefulStop()` if we want the old behavior.

These changes resulted in some collateral changes to several other locations.

### CODE REVIEW

File: doc/docs/latest/contributing/windows.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 ## Install
 
+* [go v1.13.x+](https://golang.org/dl/)
```
Comment (Tryneus) [inline doc/docs/latest/contributing/windows.md]: Unrelated.

File: src/server/cmd/pachctl/cmd/cmd.go [hunk 2]
```diff
@@ -884,6 +884,5 @@ Use "{{.CommandPath}} [command] --help" for more information about a command.{{e
 		t.Funcs(templateFuncs)
 		template.Must(t.Parse(text))
 		return t.Execute(cmd.OutOrStderr(), cmd)
-		return originalUsageFunc(cmd)
 	})
```
Comment (Tryneus) [inline src/server/cmd/pachctl/cmd/cmd.go]: Unrelated.

File: src/client/pkg/grpcutil/server.go [hunk 3]
```diff
@@ -44,62 +47,90 @@ type ServerOptions struct {
...
+
+			ready <- true
+
+			return grpcServer.Serve(listener)
+		})
 	}
-	return nil
+
+	for range servers {
+		<-ready
```
Comment (jdoliner) [inline src/client/pkg/grpcutil/server.go]: I think this needs to be a select on `ready` and `ctx.Done()` so in case the ctx gets cancelled during the startup process it doesn't deadlock. Not 100% sure, because it looks like the `ctx` might not actually be used until after the ready signal is sent. I'd add it anyway though just to be defensive, I could see this code changing and adding something that touches the context.
